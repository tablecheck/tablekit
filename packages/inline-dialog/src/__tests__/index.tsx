import { mount } from '@tablecheck/tablekit-enzyme';

import { InlineDialog } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

describe('Inline Dialog', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(
      <InlineDialog
        renderTrigger={({ onClick, isOpen, ref }) => (
          <button type="button" ref={ref} onClick={onClick}>
            {isOpen ? 'open' : 'close'}
          </button>
        )}
      >
        test
      </InlineDialog>
    );
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(
      mount(
        <InlineDialog
          renderTrigger={({ onClick, isOpen, ref }) => (
            <button type="button" ref={ref} onClick={onClick}>
              {isOpen ? 'open' : 'close'}
            </button>
          )}
        >
          test
        </InlineDialog>
      )
    ).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(
      mount(
        <InlineDialog
          className="test-class"
          renderTrigger={({ onClick, isOpen, ref }) => (
            <button type="button" ref={ref} onClick={onClick}>
              {isOpen ? 'open' : 'close'}
            </button>
          )}
        >
          test
        </InlineDialog>
      )
    ).toMatchSnapshot();
  });
});
