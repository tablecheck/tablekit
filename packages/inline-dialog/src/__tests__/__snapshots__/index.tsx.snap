// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline Dialog should allow passing of custom className 1`] = `
.emotion-0 {
  position: relative;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}

<InlineDialog
  className="test-class"
  renderTrigger={[Function]}
>
  <InlineDialogContainer
    className="test-class"
  >
    <div
      className="test-class emotion-0 emotion-1"
    >
      <Manager>
        <Reference>
          <button
            onClick={[Function]}
            type="button"
          >
            close
          </button>
        </Reference>
      </Manager>
    </div>
  </InlineDialogContainer>
</InlineDialog>
`;

exports[`Inline Dialog should correctly apply styling 1`] = `
.emotion-0 {
  position: relative;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}

<InlineDialog
  renderTrigger={[Function]}
>
  <InlineDialogContainer>
    <div
      className="emotion-0 emotion-1"
    >
      <Manager>
        <Reference>
          <button
            onClick={[Function]}
            type="button"
          >
            close
          </button>
        </Reference>
      </Manager>
    </div>
  </InlineDialogContainer>
</InlineDialog>
`;
